src/kernel/sporadic.c

```diff
@@ -303,17 +303,18 @@ void refill_unblock_check(sched_context_t *sc)
     /* advance earliest activation time to now */
     REFILL_SANITY_START(sc);
     if (refill_ready(sc)) {
-        refill_head(sc)->rTime = NODE_STATE_ON_CORE(ksCurTime, sc->scCore);
+        refill_head(sc)->rTime = NODE_STATE_ON_CORE(ksCurTime, sc->scCore) + getKernelWcetTicks();
```

Comment:
Could this alternatively be `MAX(NODE_STATE_ON_CORE(ksCurTime, sc->scCore), refill_head(sc)->rTime)`?

Reply:
Is there a reason `refill_ready` cannot be changed to return `REFILL_HEAD(sc).rTime <= NODE_STATE_ON_CORE(ksCurTime, sc->scCore)`?

Reply:
It appears that originally an SC could get released WCET time early allowing the budget to be exceeded.  And now it appears that the first WCET time that an SC runs for after being unblocked is potentially going to not be accounted for?

Reply:
Previously, both the SC would be released WCET early but the timeout for exhaustion would be set based on the release time in the head rather than by the actual time it was released which could lead to overrun. I don't think what we have here is a reasonably solution anyway, we should probably only release tasks early by the timer latency (if at all) and account for the early release by setting the timeout earlier (which is already happening to some degree). https://github.com/xurtis/seL4/blob/mcs-changes/src/kernel/thread.c#L572-L586

I thought I had made some changes to this to be more selective about which timer events get triggered early, they appear to have gotten lost. :/

Reply:
I need to take another look at this commit and figure out what the actual issue is here and how to best address it.

Reply:
I'm going to drop this change, I don't think it is correct. I'm still concerned that considering a task to be 'ready' early by as much a the kernel WCET is going to effectively reduce the period of all tasks by the kernel WCET which seems undesirable.